> Concerning the point about php-embed, there was a comment in the derivation code that said that uWSGI needed php to contain the configure paths. I have built uwsgi with php using my patch, and am running it without any problems, so perhaps it is no longer the case that uWSGI needs the configure info.

That comment was mine - I tried your PR and I get these compile errors (again):
```log
*** uWSGI is ready, launch it with ./uwsgi ***
using profile: buildconf/nixos.ini
detected include path: ['/nix/store/p44wyjqcmb8ldjj3yxjf9v3d4ml6k29x-gcc-5.4.0/lib/gcc/x86_64-unknown-linux-gnu/5.4.0/include', '/nix/store/p44wyjqcmb8ldjj3yxjf9v3d4ml6k29x-gcc-5.4.0/include', '/nix/store/p44wyjqcmb8ldjj3yxjf9v3d4ml6k29x-gcc-5.4.0/lib/gcc/x86_64-unknown-linux-gnu/5.4.0/include-fixed', '/nix/store/hrgvzzs1j58cr0ngb7awjxad1lp2zrqh-glibc-2.25-dev/include']
*** uWSGI building and linking plugin plugins/python ***
[gcc] /nix/store/6d8p7gi1603lqp5ahrly5l08ldzhjix7-uwsgi-2.0.14/lib/uwsgi/python3_plugin.so
build time: 7 seconds
*** python3 plugin built and available in /nix/store/6d8p7gi1603lqp5ahrly5l08ldzhjix7-uwsgi-2.0.14/lib/uwsgi/python3_plugin.so ***
/nix/store/9nr5wrxaiw43f7a6rgj3xxz5gzkf4yvy-php-7.1.2/bin/php-config: line 20: out: command not found
/nix/store/9nr5wrxaiw43f7a6rgj3xxz5gzkf4yvy-php-7.1.2/bin/php-config: line 20: out: command not found
/nix/store/9nr5wrxaiw43f7a6rgj3xxz5gzkf4yvy-php-7.1.2/bin/php-config: line 20: out: command not found
/nix/store/9nr5wrxaiw43f7a6rgj3xxz5gzkf4yvy-php-7.1.2/bin/php-config: line 20: out: command not found
In file included from plugins/php/php_plugin.c:1:0:
plugins/php/common.h:1:17: fatal error: php.h: No such file or directory
compilation terminated.
In file included from plugins/php/session.c:1:0:
plugins/php/common.h:1:17: fatal error: php.h: No such file or directory
compilation terminated.
using profile: buildconf/nixos.ini
detected include path: ['/nix/store/p44wyjqcmb8ldjj3yxjf9v3d4ml6k29x-gcc-5.4.0/lib/gcc/x86_64-unknown-linux-gnu/5.4.0/include', '/nix/store/p44wyjqcmb8ldjj3yxjf9v3d4ml6k29x-gcc-5.4.0/include', '/nix/store/p44wyjqcmb8ldjj3yxjf9v3d4ml6k29x-gcc-5.4.0/lib/gcc/x86_64-unknown-linux-gnu/5.4.0/include-fixed', '/nix/store/hrgvzzs1j58cr0ngb7awjxad1lp2zrqh-glibc-2.25-dev/include']
*** uWSGI building and linking plugin plugins/php ***
[gcc] /nix/store/6d8p7gi1603lqp5ahrly5l08ldzhjix7-uwsgi-2.0.14/lib/uwsgi/php_plugin.so
*** unable to build php plugin ***
```

I remember that I tracked the issue of the missing `php.h` down to uwsgi trying to access the libraries PHP was built with and failing because of the paths being unset. I spent a few hours trying to fix and could not find a different solution 😃

--- a/pkgs/development/interpreters/php/default.nix
+++ b/pkgs/development/interpreters/php/default.nix
@@ -23,11 +23,6 @@ let
       buildInputs = [ flex bison pkgconfig ]
         ++ lib.optional stdenv.isLinux systemd;
 
-      configureFlags = [
-        "EXTENSION_DIR=$(out)/lib/php/extensions"
-      ] ++ lib.optional stdenv.isDarwin "--with-iconv=${libiconv}"
-        ++ lib.optional stdenv.isLinux  "--with-fpm-systemd";
-
       flags = {
 
         # much left to do here...
@@ -267,24 +262,38 @@ let
 
       hardeningDisable = [ "bindnow" ];
 
-      configurePhase = ''
+      preConfigure = ''
         # Don't record the configure flags since this causes unnecessary
-        # runtime dependencies - except for php-embed, as uwsgi needs them.
-        ${lib.optionalString (!(config.php.embed or false)) ''
+        # runtime dependencies
         for i in main/build-defs.h.in scripts/php-config.in; do
           substituteInPlace $i \
             --replace '@CONFIGURE_COMMAND@' '(omitted)' \
             --replace '@CONFIGURE_OPTIONS@' "" \
             --replace '@PHP_LDFLAGS@' ""
         done
-        ''}
 
-        [[ -z "$libxml2" ]] || export PATH=$PATH:$libxml2/bin
-        ./configure --with-config-file-scan-dir=/etc/php.d --with-config-file-path=$out/etc --prefix=$out $configureFlags
+        #[[ -z "$libxml2" ]] || addToSearchPath PATH $libxml2/bin
+
+        configureFlags+=(--with-config-file-path=$out/etc \
+          --includedir=$dev/include \
+          EXTENSION_DIR=$out/lib/php/extensions)
       '';
+
+      configureFlags = [
+        "--with-config-file-scan-dir=/etc/php.d"
+      ] ++ lib.optional stdenv.isDarwin "--with-iconv=${libiconv}"
+        ++ lib.optional stdenv.isLinux  "--with-fpm-systemd";
 
       postInstall = ''
         cp php.ini-production $out/etc/php.ini
+      '';
+
+      postFixup = ''
+        mkdir -p $dev/bin $dev/share/man/man1
+        mv $out/bin/phpize $out/bin/php-config $dev/bin/
+        mv $out/share/man/man1/phpize.1.gz \
+          $out/share/man/man1/php-config.1.gz \
+          $dev/share/man/man1/
       '';
 
       src = fetchurl {
@@ -298,6 +307,7 @@ let
         license = licenses.php301;
         maintainers = with maintainers; [ globin ];
         platforms = platforms.all;
+        outputsToInstall = [ "out" "dev" ];
       };
 
       patches = if !php7 then [ ./fix-paths.patch ] else [ ./fix-paths-php7.patch ];

